flink-cdc-e2e-tests/flink-cdc-pipeline-e2e-tests/src/test/java/org/apache/flink/cdc/pipeline/tests/utils/PipelineTestEnvironment.java

```diff
@@ -60,16 +60,16 @@
 
 import static org.apache.flink.util.Preconditions.checkState;
 
-/** Test environment running pipeline job on Flink containers. */
 @RunWith(Parameterized.class)
 public abstract class PipelineTestEnvironment extends TestLogger {
     private static final Logger LOG = LoggerFactory.getLogger(PipelineTestEnvironment.class);
 
-    @Parameterized.Parameter public String flinkVersion;
+    @Parameterized.Parameter(0)
+    public String flinkVersion;
+
+    @Parameterized.Parameter(1)
+    public String elasticsearchVersion;
```

Comment:
Putting Elasticsearch specific things in `PipelineTestEnvironment` seems not a good idea. Other parameterized E2e cases that has nothing to do with ES will be executed repeatedly, which doesn't make sense.

Reply:
Please rebase it with `master` since there's some conflicts in this file.